src/LinAlg/hiopLinSolverSparseCUSOLVER.cpp

```diff
@@ -473,13 +473,10 @@ namespace hiop
                                         d_work_);
       //end of factor
     }    
-    nlp_->runStats.linsolv.tmInertiaComp.start();
-    int negEigVal = nFakeNegEigs_;
-    nlp_->runStats.linsolv.tmInertiaComp.stop();
-    return negEigVal;
+    return 0;
```

Comment:
I don't think `sp_status_` can return info on matrix singularity. Here are available [error messages](https://docs.nvidia.com/cuda/cusolver/index.html#cuSolverSPstatus).

Reply:
> I don't think `sp_status_` can return info on matrix singularity. Here are available [error messages](https://docs.nvidia.com/cuda/cusolver/index.html#cuSolverSPstatus).

the factorization routine should return some error code when at least when a zero pivot is encountered. Check the user manual for the functions, should have a "info" parameter ... or there may be a function that checks for zero pivots, like the cuSolver Cholesky API offers.

Reply:
Based on other similar cuSOLVER types, I assume:
1. `csrgluInfo`, the type of `*info_M_` is an `enum`.
2. 0 is good, otherwise bad.

With these assumptions, I modified the method `matrixChanged` to return -1 if the `info_M_` suggests matrix factorization was not successful.